crates/ra_ide/src/completion/complete_qualified_path.rs

```diff
@@ -584,6 +572,39 @@ mod tests {
         );
     }
 
+    #[test]
+    fn completes_trait_associated_type() {
+        assert_debug_snapshot!(
+            do_reference_completion(
+                r"
+                mod foo {
+                    pub struct S;
+                }
+                trait Tr {
+                    type Assoc;
+                }
+                impl Tr for foo::S {}
+
+                fn f() {
+                    foo::S::<|>
+                }
+                ",
+            ),
+            @r###"
+        [
+            CompletionItem {
+                label: "Assoc",
```

Comment:
... so this completion is actually wrong, rustc will [complain about this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9958c888779ff6c5bc5f2c352e208cca).

Reply:
Oh great, not sure how I missed that :/

I think this is actually a bug in rustc, namely https://github.com/rust-lang/rust/issues/22519. But yeah, rust-analyzer shouldn't suggest something that doesn't work.

Reply:
As far as I can see, that issue doesn't propose allowing `foo::S::AssocTy`. It's rather about the fact that even if `T` is a type parameter, `<T>::AssocTy` doesn't work.